(GH-5) feat: Added live user location to SGW and Loyola maps.

NekZampe opened the pull request:
# Objectives of the Pull Request ? 
Added live user location on to SGW and Loyola maps, which includes a toggle to turn on and off and a button to re-center to location. User can now see when they are in or around a certain building. 

Note: Updated .gitignore to ignore Android studio IDE specific files such as: ".idea/"

# What is left out of the Pull Request ? 
Later we'll need to add a UI toggle to center and follow user location on map ( currently toggled off in settings ).

# How to run the pull request ?
 
```bash
# Assuming you are in a folder containing minicap_concordia_campus_guide_app/application folder.
# Update the code
git checkout feat/live_user_location
git pull
# Might be required if you update check the PR

# Run npx
npx expo start
```

# Checklist
> Please, read carefully each item before checking it. Your PR's review might be delayed otherwise.

* **Code** :
  * [x] The code I'm asking a review for is working. **I understand that my PR will be rejected as-is otherwise.**
  * [x] My code follows the conventions described in the `contribution guide`.
 
* **Documentation** : 
  * [ ]  I have updated the documentation (README) accordingly to my changes.
 
* **Pull-request** : 
  * [x]  My merge message is prefixed with a `commitizen` key word so it looks something like `<feat|fix|chore|doc|refactor|perf|style>: foo bar`**
  * [x]  The code *provided to run the pull request* is working. **I understand that my PR will be rejected as-is otherwise.**
  * [x]  My pull request is documented. I have explained the needs for the PR, what was left out of the it and why.
  * [x]  I have carefully reviewd each changes made to a file and made sure the files included on the PR were actually added on purpose.
  * [x]  I have assignee at least **1** people to review my PR.


Link on documentation for setting up current location on your emulator in android studio:
https://docs.google.com/document/d/1PFqj28MRtymIoxlB3C_UQmKkBgDBS5k3QZHpHX09_dY/edit?usp=sharing

NekZampe committed:
Added live user location to SGW and Loyola maps

AlimuratDinch commented:
@NekZampe The section on `How to run your pull request` should be completed with the branch name that we want to test

NekZampe commented:
> @NekZampe The section on `How to run your pull request` should be completed with the branch name that we want to test

ok I updated it

AlimuratDinch commented on application/package.json:
```diff
@@ -46,6 +46,7 @@
     "expo-font": "~13.0.3",
     "expo-haptics": "~14.0.1",
     "expo-linking": "~7.0.5",
+    "expo-location": "^18.0.5",
```
@NekZampe Since there is already expo-location library, we do not need to install expo-location again (what is not written on the Pull Request description). But it is not really an issue `expo-location`  works good!

NekZampe replied:
ah ok I can remove that requirement from the pull request

AlimuratDinch reviewed (commented):
@NekZampe 
- It does not show me the location am currently in. Maybe it is because the android emulator app cannot see my location. So i would need someone else to confirm that it shows the current location
- But the rest. LGTM, Once there is someone who can prove that the functionning is working, i can approve this PR
- ![image](https://github.com/user-attachments/assets/4641ec38-0f9d-4990-9e23-1c638abfc376)

NekZampe commented:
> @NekZampe
> 
> * It does not show me the location am currently in. Maybe it is because the android emulator app cannot see my location. So i would need someone else to confirm that it shows the current location
> * But the rest. LGTM. Once there is someone who can prove that the functionning is working, i can approve this PR
> * ![image](https://private-user-images.githubusercontent.com/132238708/410724784-4641ec38-0f9d-4990-9e23-1c638abfc376.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4OTQ1ODEsIm5iZiI6MTczODg5NDI4MSwicGF0aCI6Ii8xMzIyMzg3MDgvNDEwNzI0Nzg0LTQ2NDFlYzM4LTBmOWQtNDk5MC05ZTIzLTFjNjM4YWJmYzM3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwN1QwMjExMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YWRkZDViZDM0ZjhlZDk5MGY4NDM3MTc1ZDdhMWZiYTViMDZiMjM2Mjg5ZjgwMzUzNmI5Y2FmZmM5NTgwZTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Yc1qTMaxtkoeqdo1FzQrA_u05-aU47dSFgZ5Aol_3YU)

I added documentation on setting up location on android emulator in android studio

AlimuratDinch commented:
> > @NekZampe
> > 
> > * It does not show me the location am currently in. Maybe it is because the android emulator app cannot see my location. So i would need someone else to confirm that it shows the current location
> > * But the rest. LGTM. Once there is someone who can prove that the functionning is working, i can approve this PR
> > * ![image](https://private-user-images.githubusercontent.com/132238708/410724784-4641ec38-0f9d-4990-9e23-1c638abfc376.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4OTQ1ODEsIm5iZiI6MTczODg5NDI4MSwicGF0aCI6Ii8xMzIyMzg3MDgvNDEwNzI0Nzg0LTQ2NDFlYzM4LTBmOWQtNDk5MC05ZTIzLTFjNjM4YWJmYzM3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwN1QwMjExMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YWRkZDViZDM0ZjhlZDk5MGY4NDM3MTc1ZDdhMWZiYTViMDZiMjM2Mjg5ZjgwMzUzNmI5Y2FmZmM5NTgwZTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Yc1qTMaxtkoeqdo1FzQrA_u05-aU47dSFgZ5Aol_3YU)
> 
> I added documentation on setting up location on android emulator in android studio: https://docs.google.com/document/d/1PFqj28MRtymIoxlB3C_UQmKkBgDBS5k3QZHpHX09_dY/edit?usp=sharing
> 
> This is for anyone needing to test this part, let me know if you have any issues.

Perfect, thank you. I sent the request to access the document

NekZampe commented:
> > > @NekZampe
> > > 
> > > * It does not show me the location am currently in. Maybe it is because the android emulator app cannot see my location. So i would need someone else to confirm that it shows the current location
> > > * But the rest. LGTM. Once there is someone who can prove that the functionning is working, i can approve this PR
> > > * ![image](https://private-user-images.githubusercontent.com/132238708/410724784-4641ec38-0f9d-4990-9e23-1c638abfc376.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4OTQ1ODEsIm5iZiI6MTczODg5NDI4MSwicGF0aCI6Ii8xMzIyMzg3MDgvNDEwNzI0Nzg0LTQ2NDFlYzM4LTBmOWQtNDk5MC05ZTIzLTFjNjM4YWJmYzM3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwN1QwMjExMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YWRkZDViZDM0ZjhlZDk5MGY4NDM3MTc1ZDdhMWZiYTViMDZiMjM2Mjg5ZjgwMzUzNmI5Y2FmZmM5NTgwZTNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Yc1qTMaxtkoeqdo1FzQrA_u05-aU47dSFgZ5Aol_3YU)
> > 
> > 
> > I added documentation on setting up location on android emulator in android studio: https://docs.google.com/document/d/1PFqj28MRtymIoxlB3C_UQmKkBgDBS5k3QZHpHX09_dY/edit?usp=sharing
> > This is for anyone needing to test this part, let me know if you have any issues.
> 
> Perfect, thank you. I sent the request to access the document

I updated it so anyone with the link can view, thank you!

AlimuratDinch reviewed (commented):
LGTM. NIce job! I have added the link in our `wikis`
![image](https://github.com/user-attachments/assets/040a4f86-2f63-4399-8a6b-42acbe83e415)

JosephAladas commented:
<img width="380" alt="Screenshot 2025-02-07 at 9 10 09 PM" src="https://github.com/user-attachments/assets/a007fbed-41cc-4769-a0e8-dc58594cfecb" />

JosephAladas reviewed (approved):
LGTM
